### PR TITLE
 [FLINK-20209][web] Add tolerable failed checkpoints config to web ui

### PR DESCRIPTION
**Pull request by coolderli**
## What is the purpose of the change

- Add `Tolerable Failed Checkpoints` to checkpoint configuration in Flink UI.

## Brief change log

- Add tolerableFailed config to REST endpoint, and web ui.

## Verifying this change

- This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no) No
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no) No
  - The serializers: (yes / no / don't know) No
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know) No
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / no / don't know) No
  - The S3 file system connector: (yes / no / don't know) No

## Documentation

  - Does this pull request introduce a new feature? (yes / no) No
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by coolderli** [comment]
image

**Comment by Myasuka** [comment]
@coolderli Have you ever checked the failure test `RestAPIStabilityTest.testDispatcherRestAPIStability` in last [CI](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=10513&view=logs&j=0da23115-68bb-5dcd-192c-bd4c8adebde1&t=05b74a19-4ee4-5036-c46f-ada307df6cf0), did it has relationship with your PR?

**Comment by coolderli** [comment]
@flinkbot run travis

**Comment by coolderli** [comment]
@flinkbot run azure

**Comment by coolderli** [comment]
image

**Comment by coolderli** [comment]
@Myasuka Fix [ci](https://dev.azure.com/apache-flink/98463496-1af2-4620-8eab-a2ecc1a2e6fe/_build/results?buildId=10739), please review the code, the effect picture is above, Thanks.

File: flink-runtime-web/src/test/resources/rest_api_v1.snapshot [hunk 1]
```diff
@@ -1242,6 +1242,9 @@
         },
         "unaligned_checkpoints" : {
           "type" : "boolean"
+        },
+        "tolerable_failed" : {
```
Comment (Myasuka) [inline flink-runtime-web/src/test/resources/rest_api_v1.snapshot:1246]: how about `tolerable_failed_checkpoints` ?

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointConfigInfo.java [hunk 2]
```diff
@@ -57,6 +57,8 @@
 
 	public static final String FIELD_NAME_UNALIGNED_CHECKPOINTS = "unaligned_checkpoints";
 
+	public static final String FIELD_NAME_TOLERABLE_FAILED_CHECKPOINT = "tolerable_failed";
```
Comment (Myasuka) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointConfigInfo.java:60]: how about changes below?
```suggestion
	public static final String FIELD_NAME_TOLERABLE_FAILED_CHECKPOINTS = "tolerable_failed_checkpoints";
```

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointConfigInfo.java [hunk 3]
```diff
@@ -81,6 +83,9 @@
 	@JsonProperty(FIELD_NAME_UNALIGNED_CHECKPOINTS)
 	private final boolean unalignedCheckpoints;
 
+	@JsonProperty(FIELD_NAME_TOLERABLE_FAILED_CHECKPOINT)
+	private final int tolerableFailed;
```
Comment (Myasuka) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointConfigInfo.java:87]: The same suggestion as above.

**Review by Myasuka — COMMENTED** [review]
Leave comments of some naming.
BTW, I noticed that you have included another `XIAOMIlipeidian` id in the commit, please consider to use one as author.

**Comment by coolderli** [comment]
@flinkbot run azure

**Comment by Myasuka** [comment]
@coolderli I noticed that the latest PR still combines two git user ids: `coolderli` and `XIAOMIlipeidian` as I pointed out in last comment. Which user id do you prefer to count as contribution? Please consider to re-push the commit with correct information.

**Comment by zentol** [comment]
The REST API documentation needs to be updated (the .snapshot file just for testing the API stability).

**Comment by coolderli** [comment]
@flinkbot  run azure

**Comment by coolderli** [comment]
> @coolderli I noticed that the latest PR still combines two git user ids: `coolderli` and `XIAOMIlipeidian` as I pointed out in last comment. Which user id do you prefer to count as contribution? Please consider to re-push the commit with correct information.

Thank you for reminding me. That's because I configured the global user.email. Now I've fixed it.

**Comment by coolderli** [comment]
> The REST API documentation needs to be updated (the .snapshot file just for testing the API stability). (see https://github.com/apache/flink/tree/master/flink-docs#rest-api-documentation)

Now it's updated. Thank you！

**Review by Myasuka — APPROVED** [review]
LGTM

**Pull request merged**